src/snapred/backend/data/LocalDataService.py

```diff
@@ -535,292 +340,155 @@ def writeNormalizationRecord(
-        #   to correspond to the filenames that will actually be saved to disk.
-        savedWorkspaces = {}
-        workspaces = record.workspaces.copy()
-        wss = []
-        for wsName in workspaces.pop(wngt.DIFFCAL_OUTPUT, []):
-            # Rebuild the workspace name to strip any "iteration" number:
-            #   * WARNING: this workaround does not work correctly if there are multiple workspaces of each "unit" type.
-            if wng.Units.DSP.lower() in wsName:
-                ws = (
-                    wng.diffCalOutput()
-                    .unit(wng.Units.DSP)
-                    .runNumber(record.runNumber)
-                    .version(record.version)
-                    .group(record.focusGroupCalibrationMetrics.focusGroupName)
-                    .build()
-                )
-            else:
-                raise RuntimeError(
-                    f"cannot save a workspace-type: {wngt.DIFFCAL_OUTPUT} without a units token in its name {wsName}"
-                )
-            wss.append(ws)
-        savedWorkspaces[wngt.DIFFCAL_OUTPUT] = wss
-        wss = []
-        for wsName in workspaces.pop(wngt.DIFFCAL_DIAG, []):
-            # Rebuild the workspace name to strip any "iteration" number:
-            #   * WARNING: this workaround does not work correctly if there are multiple workspaces of each "unit" type.
-            if wng.Units.DIAG.lower() in wsName:
-                ws = (
-                    wng.diffCalOutput()
-                    .unit(wng.Units.DIAG)
-                    .runNumber(record.runNumber)
-                    .version(record.version)
-                    .group(record.focusGroupCalibrationMetrics.focusGroupName)
-                    .build()
-                )
-            else:
-                raise RuntimeError(
-                    f"cannot save a workspace-type: {wngt.DIFFCAL_DIAG} without a units token in its name {wsName}"
-                )
-            wss.append(ws)
-        savedWorkspaces[wngt.DIFFCAL_DIAG] = wss
-        wss = []
-        for wsName in workspaces.pop(wngt.DIFFCAL_TABLE, []):
-            # Rebuild the workspace name to strip any "iteration" number:
-            #   * WARNING: this workaround does not work correctly if there are multiple table workspaces.
-            ws = wng.diffCalTable().runNumber(record.runNumber).version(record.version).build()
-            wss.append(ws)
-        savedWorkspaces[wngt.DIFFCAL_TABLE] = wss
-        wss = []
-        for wsName in workspaces.pop(wngt.DIFFCAL_MASK, []):
-            # Rebuild the workspace name to strip any "iteration" number:
-            #   * WARNING: this workaround does not work correctly if there are multiple mask workspaces.
-            ws = wng.diffCalMask().runNumber(record.runNumber).version(record.version).build()
-            wss.append(ws)
-        savedWorkspaces[wngt.DIFFCAL_MASK] = wss
-        savedRecord = deepcopy(record)
-        savedRecord.workspaces = savedWorkspaces
```

Comment:
All of this logic was being duplicated, first for saving records and secondly for saving workspaces.  It was moved to a single space, inside the `CalibrationService`.

Reply:
Good.  However, please double check that `Normalization` service is also doing this?  (Possibly I was misunderstanding some of the previous changes...  :)  )